### PR TITLE
Extend ext_authz fuzzer coverage to grpc client

### PR DESCRIPTION
**Pull request by antoniovleonti**
Commit Message: Extend ext_authz fuzzer coverage to grpc client
Additional Description: Previously, the ext_authz fuzzer just used a mock client. This PR replaces the mock client with a real grpc client (which makes mock grpc calls) to extend the fuzzer coverage. 
This PR also removes the static qualifier of the `mocks` object as this increases the cognitive load of making changes to the fuzzer.
Risk Level: None
Testing: Test-only change
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: test/extensions/filters/http/ext_authz/ext_authz_fuzz_test.cc [hunk 1]
```diff
@@ -92,7 +89,7 @@ DEFINE_PROTO_FUZZER(const envoy::extensions::filters::http::ext_authz::ExtAuthzT
     return;
   }
 
-  static FuzzerMocks mocks;
+  FuzzerMocks mocks;
```
Comment (yanjunxiang-google) [inline test/extensions/filters/http/ext_authz/ext_authz_fuzz_test.cc:92]: Thanks for working on this!
A few high nit comments first: Is this change intentional?

File: test/extensions/filters/http/ext_authz/ext_authz_fuzz_test.cc [hunk 2]
```diff
@@ -110,32 +107,45 @@ DEFINE_PROTO_FUZZER(const envoy::extensions::filters::http::ext_authz::ExtAuthzT
...
+                << "Could not parse serialized check request";
+
+            // TODO: Query the request header map in HttpFilterFuzzer to test
+            // headers_to_(add/remove/append).
+            // TODO: Test check request attributes against config
+            // and filter metadata.
+            ENVOY_LOG_MISC(trace, "Check Request attributes {}",
+                           check_request.attributes().DebugString());
+
+            grpc_client->onSuccess(makeCheckResponse(resultCaseToCheckStatus(input.result())),
```
Comment (yanjunxiang-google) [inline test/extensions/filters/http/ext_authz/ext_authz_fuzz_test.cc:139]: Should we also cover onFailure() case?

**Review by ggreenway — APPROVED** [review]
LGTM modulo comment from @yanjunxiang-google 

/wait-any

File: test/extensions/filters/http/ext_authz/ext_authz_fuzz_test.cc (hunk 1 above)
Comment (antoniovleonti) [reply test/extensions/filters/http/ext_authz/ext_authz_fuzz_test.cc]: Yes, but I realized the real problem is just the stateful mocks being static. I've refactored a bit. Hopefully my intent is more clear.

File: test/extensions/filters/http/ext_authz/ext_authz_fuzz_test.cc (hunk 2 above)
Comment (antoniovleonti) [reply test/extensions/filters/http/ext_authz/ext_authz_fuzz_test.cc]: Yes, done.

**Review by yanjunxiang-google — APPROVED** [review]

**Comment by antoniovleonti** [comment]
@ggreenway I think it might be waiting on you now?

**Review by ggreenway — DISMISSED** [review]
LGTM, but there's a merge conflict

/wait

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Comment by adisuissa** [comment]
This PR was approved by Greg, so enabling auto-merge.
